pkg/server/options/authorization.go

```diff
@@ -101,24 +101,25 @@ func (s *Authorization) ApplyTo(config *genericapiserver.Config, informer kcpkub
 	}
 
 	// kcp authorizers
-	bootstrapAuth, bootstrapRules := authorization.NewBootstrapPolicyAuthorizer(informer)
-	localAuth, localResolver := authorization.NewLocalAuthorizer(informer)
-	apiBindingAuth, err := authorization.NewMaximalPermissionPolicyAuthorizer(informer, kcpinformer,
-		union.New(bootstrapAuth, localAuth),
-	)
-	if err != nil {
-		return err
-	}
+	bootstrapAuth, bootstrapRules := authz.NewBootstrapPolicyAuthorizer(informer)
+	bootstrapAuth = authz.NewDecorator(bootstrapAuth, "bootstrap.authorization.kcp.dev").AddAuditLogging().AddAnonymization().AddReasonAnnotation()
```

Comment:
TODO: refactor so this is impossible to forget to do, and easier (I see we're also doing it in the virtual apiexport builder)

Reply:
Simple approach here would be to make the types coming out of `*.New*Authorizer()` useless without decoration

Reply:
Probably worth it to do in this PR if it's not a lot of effort

Reply:
We can also leave this in the constructor method. I moved it here because this is more plumbing. I am fine with either decision.

@stevekuznetsov good idea as well 👍 The returned type must not implement `Authorizer` though, maybe just a method name change in each authorizer, i.e. `UnsafeAuthorize(ctx, attr)` ?.

Reply:
unsafe doesn't sound good though 😆

Reply:
Would we ever want to construct a FooAuthorizer that has a different decoration prefix, doesn't have audit logging, doesn't have anon, doesn't have reason annotation?

Reply:
@ncdc probably not 🤔 that would speak for putting these things back into the constructor indeed.

Reply:
@ncdc well, yes, the very last authorizer in the delegate chain doesn't need reason annotation.

Reply:
@sttts you commit to buying us all a beer on every instance of a new authorizer not using these filters?

Reply:
For the record: on next KubeCon where we both meet, @stevekuznetsov will get a beer per authorizer until then that does not use the filters.